util/src/common.rs

```diff
@@ -25,6 +25,20 @@ pub use numbers::*;
 pub use sha3::*;
 
 #[macro_export]
+macro_rules! hash_map {
+	( $( $x:expr => $y:expr ),* ) => {
+		vec![ $( ($x, $y) ),* ].into_iter().collect::<HashMap<_, _>>()
+	}
+}
+
+#[macro_export]
+macro_rules! hash_mapx {
+	( $( $x:expr => $y:expr ),* ) => {
+		vec![ $( ( From::from($x), From::from($y) ) ),* ].into_iter().collect::<HashMap<_, _>>()
```

Comment:
Could probably be done with a simple array instead of vector (also applies to similar macross below)